qiskit/algorithms/evolvers/real/trotterization/trotter_ops_validator.py

```diff
@@ -0,0 +1,131 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Set of method for validating input to TrotterQrte algorithm."""
+
+import numbers
+from typing import Union
+
+from qiskit.circuit import Parameter, ParameterExpression
+from qiskit.opflow import (
+    SummedOp,
+    PauliOp,
+)
+
+
+def is_op_bound(operator: Union[SummedOp, PauliOp]) -> None:
+    """Checks if an operator provided has all parameters bound.
+
+    Args:
+        operator: Operator to be checked.
+
+    Raises:
+        ValueError: If an operator has unbound parameters.
+    """
+    if len(operator.parameters) > 0:
+        raise ValueError(
+            f"Did not manage to bind all parameters in the Hamiltonian, "
+            f"these parameters encountered: {operator.parameters}."
+        )
+
+
+def validate_hamiltonian_form(hamiltonian: Union[SummedOp, PauliOp]):
+    """Validates that a Hamiltonian is of a correct type and with expected dependence on
+    parameters.
+
+    Args:
+        hamiltonian: Hamiltonian to be validated.
+
+    Raises:
+        ValueError: if an invalid Hamiltonian is provided.
+    """
+    if isinstance(hamiltonian, SummedOp):
```

Comment:
In the error messages here it would be nice to include exactly what caused the error into the formatted message. It can be informative/helpful to see why it went wrong specifically.